### ISSUE TITLE
Implement State save restore to disk on SIGINT to allow to stop and resume a fuzzer

### ISSUE THREAD

**Issue by andreafioraldi** [AFLplusplus/LibAFL#288]
We should implement the handler for posix and windows and extend it to work with multiple processes when using a launcher

**Comment by domenukk** [comment]
I don't think we have the state available at any time, necessarily. We could set a "exit_soon" flag to true in the handler and check that at certain points in time

**Comment by domenukk** [comment]
An AFL-Like resume with ondiskcorpus may be fine, though?

**Comment by s1341** [comment]
Currently loading from an OnDiskCorpus is quite horrific, especially when running ('resuming') on multiple cores. It takes forever and each core sends all testcases to every other core.... We need to provide a better user experience for this.

**Comment by s1341** [comment]
After running a fuzz-campaign for about a week with LibAFL, it's clear to me that some way to stop and resume the fuzzing, potentially with tweaks to the fuzzer in between, is a critical requirement.

Assuming we just re-read the queue, we need to either delegate the queue-reading to a single node (a 'queue importer'?), or find a way to efficiently spread the queue-reading across nodes, without causing a cascade of 'new test-cases' from every node to every other node.

**Comment by tokatoka** [comment]
I think at least we should add proper signal handlings for SIGINT and SIGTERM

**Comment by tokatoka** [comment]
The shmem allocated StateRestorer is not freed when the process is killed, so we need to implement this.

**Comment by domenukk** [comment]
This could still be improved but is more or less possible these days.
